libgalois/src/PropertyGraph.cpp

```diff
@@ -145,6 +146,182 @@ MakePropertyGraph(
       std::move(rdg_file), std::move(rdg_result.value()));
 }
 
+/// Assumes all boolean or uint8 properties are types
+katana::Result<std::shared_ptr<arrow::NumericArray<arrow::UInt8Type>>>
+GetTypeIDsFromProperties(
+    const std::shared_ptr<arrow::Table>& properties,
+    katana::PropertyGraph::TypeToTypeIDsMap* type_name_to_ids,
+    katana::PropertyGraph::TypeIDToTypesMap* type_id_to_names) {
+  // throw an error if each column/property has more than 1 chunk
+  for (int i = 0, n = properties->num_columns(); i < n; i++) {
+    std::shared_ptr<arrow::ChunkedArray> property = properties->column(i);
+    if (property->num_chunks() != 1) {
+      return KATANA_ERROR(
+          katana::ErrorCode::NotImplemented,
+          "execution failed because property {} has {} chunks (1 chunk "
```

Comment:
You start a bunch of error messages with `execution failed because` but that's not really meaningful since they are already marked as errors from context. I would remove that phrase throughout.